### ISSUE TITLE
cholesky.U allocates

### ISSUE THREAD

**Issue by cstjean** [JuliaArrays/StaticArrays.jl#479]
On 0.6,

```julia
julia> using StaticArrays

julia> @allocated chol(@SMatrix [1.0])
0
```

However, the recommended replacement in 0.7 allocates:

```julia
julia> using StaticArrays, LinearAlgebra

julia> @allocated cholesky(@SMatrix [1.0]).U
192
```

The allocation happens during the `.U` `getproperty` [call](https://github.com/JuliaLang/julia/blob/master/stdlib/LinearAlgebra/src/cholesky.jl#L338). I did not investigate further, but my kalman filtering code is 10X slower now, and this is a likely culprit.

**Comment by cstjean** [comment]
Might be fixed by https://github.com/JuliaLang/julia/pull/28873, since if I copy-paste that code, the allocation disappears.

**Comment by cstjean** [comment]
Fixed on 1.0.1
